daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1025,15 +1025,44 @@ private[lf] object SBuiltin {
     }
   }
 
-  /** $fetch[T]
+  final case class SBCastAnyContract(templateId: TypeConName) extends SBuiltinPure(2) {
+    override private[speedy] def executePure(args: util.ArrayList[SValue]) = {
+      def coid = getSContractId(args, 0)
+      val (actualTemplateId, record) = getSAnyContract(args, 1)
+      if (actualTemplateId != templateId) {
+        throw SErrorDamlException(IE.WronglyTypedContract(coid, templateId, actualTemplateId))
+      }
+      record
+    }
+  }
+
+  final case class SBCastAnyInterface(ifaceId: TypeConName) extends SBuiltin(3) {
```

Comment:
it doesn’t seem like you’re casting anything here. You are validating the type but in the end you set the return value to `args.get(2)` without casting it to anything else.

Reply:
To my point of view, `SBCastAnyInterface have type: Any -> IfaceId` so it actually cast something. 
even from the point of view there is an change of SValue data..

Reply:
I agree, this is a cast. Definitely worth explaining it in a comment though.

Reply:
makes sense :+1: